libs/api/domains/university-of-iceland/src/lib/universityOfIceland.resolver.ts

```diff
@@ -0,0 +1,73 @@
+import { Args, Query, Resolver } from '@nestjs/graphql'
+
+import { Inject, UseGuards } from '@nestjs/common'
+import format from 'date-fns/format'
+import is from 'date-fns/locale/is'
+
+import {
+  CurrentUser,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  UniversityOfIcelandService,
+  NemandiGetLocaleEnum,
+  NemandiFerillFerillGetLocaleEnum,
+  StudentTrackTranscripts,
+} from '@island.is/clients/university-of-iceland'
+import { ApiScope } from '@island.is/auth/scopes'
+import { StudentInfoModel } from './models/studentInfo.model'
+import { StudentInfoDetailModel } from './models/studentInfoDetail.model'
+import { GetStudentInfoDetailInput } from './dto/getStudentInfoDetail.input'
+import { GetStudentInfoInput } from './dto/getStudentInfo.input'
+import { DownloadServiceConfig } from '@island.is/nest/config'
+import type { ConfigType } from '@island.is/nest/config'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.internal)
+@Resolver()
+export class UniversityOfIcelandResolver {
+  constructor(
+    private universityOfIcelandApi: UniversityOfIcelandService,
+    @Inject(DownloadServiceConfig.KEY)
+    private readonly downloadServiceConfig: ConfigType<
+      typeof DownloadServiceConfig
+    >,
+  ) {}
+
+  @Query(() => StudentInfoModel)
+  async getStudentInfo(
+    @CurrentUser() user: User,
+    @Args('input') input: GetStudentInfoInput,
+  ): Promise<StudentTrackTranscripts> {
+    const data = await this.universityOfIcelandApi.studentInfo(
+      user,
+      input.locale as NemandiGetLocaleEnum,
+    )
+    return data
+  }
+
+  @Query(() => StudentInfoDetailModel)
+  async getStudentInfoDetail(
+    @Args('input') input: GetStudentInfoDetailInput,
+    @CurrentUser() user: User,
+  ): Promise<StudentInfoDetailModel> {
```

Comment:
`studentInfo` and `studentInfoDetail` is not in the spirit of GraphQL. Could we have this instead as a field resolver on studentInfo, eg

```
{
  universityOfIcelandStudentInfo {
    career {
      transcript
      files
      body
    }
  }
}
```

Reply:
I've updated this, is it more _spiritual_ now?